server/lib/api/transaction.js

```diff
@@ -132,7 +133,54 @@ module.exports = function transactionAPI(router) {
           }
         });
     } else if (req.query.address) {
+      getBlock(
+        { $or:
+          [
+            { 'txs.outputs.address':     req.query.address },
+            { 'txs.inputs.prevout.hash': req.query.address },
+          ],
+        },
+        { rawBlock: 0 },
+        MAX_BLOCKS,
+        (err, block) => {
+          if (err) {
+            res.status(501).send();
+            logger.log('err', err);
+          }
 
+          if (block[0]) {
```

Comment:
Agreed. This all goes away with the db api. It's messy to repeat the same checks like I'm forced to do interacting with the models directly. Building out the db api is going to remove this code pattern entirely